src/Symfony/Component/HttpKernel/DependencyInjection/ResettableServicePass.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Argument\IteratorArgument;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\HttpKernel\EventListener\ServiceResetListener;
+
+/**
+ * @author Alexander M. Turek <me@derrabus.de>
+ */
+class ResettableServicePass implements CompilerPassInterface
+{
+    const RESET_TAG = 'kernel.reset';
+
+    /**
+     * @var string
+     */
+    private $tagName;
+
+    /**
+     * @param string $tagName
+     */
+    public function __construct($tagName)
+    {
+        $this->tagName = $tagName;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $services = $methods = array();
+
+        foreach ($container->findTaggedServiceIds($this->tagName, true) as $id => $tags) {
+            $services[$id] = new Reference($id, ContainerInterface::IGNORE_ON_UNINITIALIZED_REFERENCE);
+            $methods[$id] = array();
+
+            foreach ($tags as $attributes) {
+                if (!isset($attributes['method'])) {
+                    throw new \RuntimeException(
+                        sprintf('Tag %s requires the attribute method to be set.', $this->tagName)
+                    );
+                }
+
+                $methods[$id][] = $attributes['method'];
+            }
+        }
+
+        $container->getDefinition(ServiceResetListener::class)
+            ->setArguments(array(
```

Comment:
to allow for some flexibility, this should use replaceArgument (the xml declaration will need empty placeholders for pre-declaring those: `<argument />` + `<argument type="collection" />`)